kong/clustering/control_plane.lua

```diff
@@ -356,9 +354,7 @@ function _M:check_configuration_compatibility(dp_plugin_map)
         -- CP plugin needs to match DP plugins with major version
         -- CP must have plugin with equal or newer version than that on DP
         if cp_plugin.major ~= dp_plugin.major or
-          (cp_plugin.major == dp_plugin.major and cp_plugin.minor < dp_plugin.minor) or
-          (cp_plugin.major == dp_plugin.major and cp_plugin.minor == dp_plugin.minor and
-            cp_plugin.patch < dp_plugin.patch) then
+          (cp_plugin.major == dp_plugin.major and cp_plugin.minor < dp_plugin.minor) then
```

Comment:
`cp_plugin.major == dp_plugin.major` seems redundant, since `cp_plugin.major ~= dp_plugin.major` would have short circuited this condition anyway.